packages/babel-types/src/clone/cloneNode.js

```diff
@@ -11,24 +11,29 @@ function cloneIfNode(obj, deep) {
     obj.type !== "CommentLine" &&
     obj.type !== "CommentBlock"
```

Comment:
@JLHwung because of this condition `cloneNode` will never be called on `comments` so `comment.loc` will never be null! I am adding comments to node using:
```
t.addComment(node, "leading", "leadingComments", true);
node.leadingComments[0].loc = {};
```
comment type is either `CommentLine` or `CommentBlock`!

Reply:
Good catch! I think we should short-circuit this condition when `withoutLoc` and `deep` are both true. Otherwise it is inconsistent that after `cloneWithoutLoc`, the cloned `CommentLine` still contain `loc`.

Reply:
Yes, we can use:
```
((deep && withoutLoc) ||
      (obj.type !== "CommentLine" && obj.type !== "CommentBlock"))
```
But this will new an error here:

https://github.com/babel/babel/blob/bdb4318f47ec42ca4e325b5163e2e8ad57de4849/packages/babel-types/src/clone/cloneNode.js#L50

```
Unknown node type: "CommentBlock"
```
```
Unknown node type: "CommentLine"
```

Reply:
@Taym95 Maybe we could special-case comments and have a small helper function (not exposed as part of the API) `cloneCommentWithoutLoc`?